mdal/mdal.cpp

```diff
@@ -154,16 +154,20 @@ const char *MDAL_DR_filters( DriverH driver )
 /// MESH
 ///////////////////////////////////////////////////////////////////////////////////////
 
-MeshH MDAL_LoadMesh( const char *meshFile )
+MeshH MDAL_LoadMesh( const char *uri )
 {
-  if ( !meshFile )
+  if ( !uri )
   {
     MDAL::Log::error( MDAL_Status::Err_FileNotFound, "Mesh file is not valid (null)" );
     return nullptr;
   }
 
-  std::string filename( meshFile );
-  return static_cast< MeshH >( MDAL::DriverManager::instance().load( filename ).release() );
+  std::string uriString( uri ), driver, meshFile, meshName;
+  int meshId;
+
+  MDAL::parseDriverAndMeshFromUri( uriString, driver, meshFile, meshName, meshId );
```

Comment:
just return meshName string. if driver wants to see a integer (does not support names), let each particular driver to parse string -> int and use that . For UGRID lets use mesh name and not indexes

Reply:
Well .. we were talking about specifying driver names as well

Reply:
sure, what i meant is to remove meshId agrument like this `MDAL::parseDriverAndMeshFromUri( uriString, driver, meshFile, meshName)`

Reply:
Ok, I removed meshID